src/Authenticator/CookieAuthenticator.php

```diff
@@ -97,18 +82,25 @@ public function authenticate(ServerRequestInterface $request, ResponseInterface
                 'Login credentials not found'
             ]);
         }
-        $userData = $cookies[$cookieName];
-        if ($this->getConfig('encrypt')) {
-            $userData = $this->_decrypt($userData, $this->getConfig('encryptionType'));
-        }
+        $token = $cookies[$cookieName];
+        list($username, $tokenHash) = explode(':', $token);
```

Comment:
The token could be mangled and not contain 2 parts. We should probably check that there is a `:` in the token and fail authentication early if it is not present.

Reply:
Right.

Should we encode the username as it may contain `:` character?

Reply:
Could use json_encode().